icu4c/source/test/cintltst/custrtrn.c

```diff
@@ -1554,7 +1554,7 @@ static void Test_strToJavaModifiedUTF8(void) {
     p=u_strToJavaModifiedUTF8(dest, (int32_t)sizeof(dest), &length,
                               src, UPRV_LENGTHOF(src), &errorCode);
     if( U_FAILURE(errorCode) || p!=dest ||
-        length!=UPRV_LENGTHOF(expected) || 0!=memcmp(dest, expected, length) ||
+        length!=(UPRV_LENGTHOF(expected)-1) || 0!=memcmp(dest, expected, length) ||
```

Comment:
Why do we need to NUL-terminate expected when we never look at the NUL?

Reply:
strstr expect both argument to be null terminiated, if there are no "\xc0\x80" inside expected, strstr may look at memory pass the end of the memory allocated for expected.

Reply:
It will never actually run past the end because there *is* C0 80 in that array... but I understand. tnx